internal/interceptor.go

```diff
@@ -231,6 +231,9 @@ type WorkflowOutboundInterceptor interface {
 	// NewTimer intercepts workflow.NewTimer.
 	NewTimer(ctx Context, d time.Duration) Future
 
+	// NewTimer intercepts workflow.NewTimerWithOptions.
```

Comment:
Can other primitives support metadata like activities? Do we plan to add support for them?

Reply:
Yes and yes! We intentionally decided that workflow summary/details and timer summary was the first use case. But the user metadata is on the command/event and not the attributes for this reason. Any UI/CLI-visible metadata you could ever need on an event we can look into adding.